docs/minikube-tutorial.md

```diff
@@ -0,0 +1,196 @@
+
+```
+curl https://raw.githubusercontent.com/helm/helm/master/scripts/get | bash
+```
+
+Install helm tiller:
+
+```
+helm init
+```
+
+If you have limited access to gcr.io, you can try a mirror, e.g.
+
+```
+helm init --upgrade --tiller-image registry.cn-hangzhou.aliyuncs.com/google_containers/tiller:$(helm version --client --short | grep -P -o 'v\d+\.\d+\.\d')
+```
+
+Once it is installed, running `helm version` should show you both the client
+and server version, e.g.
+
+```
+$ helm version
+Client: &version.Version{SemVer:"v2.13.1",
+GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
+Server: &version.Version{SemVer:"v2.13.1",
+GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
+```
+
+If it shows only the client version, `helm` cannot yet connect to the server. Use
+`kubectl` to see if any tiller pods are running.
+
+### Install TiDB operator in the Kubernetes cluster
+
+Clone tidb-operator repository:
+
+```
+git clone https://github.com/pingcap/tidb-operator
+cd tidb-operator
+kubectl apply -f ./manifests/crd.yaml
+helm install charts/tidb-operator --name tidb-operator --namespace tidb-admin
+```
+
+Now, we can watch the operator come up with:
+
+```
+watch kubectl get pods --namespace tidb-admin -o wide
+```
+
+If you have limited access to gcr.io (pods failed with ErrImagePull), you can
+try a mirror of kube-scheduler image. You can upgrade tidb-operator like this:
+
+```
+helm upgrade tidb-operator charts/tidb-operator --namespace tidb-admin --set \
+  scheduler.kubeSchedulerImageName=registry.cn-hangzhou.aliyuncs.com/google_containers/kube-scheduler
+```
+
+When you see both tidb-scheduler and tidb-controller-manager are running, you
+can process to launch a TiDB cluster!
+
+### Launch a TiDB cluster
+
+```
+helm install charts/tidb-cluster --name tidb-cluster --set \
+  schedulerName=default-scheduler,pd.storageClassName=standard,tikv.storageClassName=standard,pd.replicas=1,tikv.replicas=1,tidb.replicas=1
+```
+
+Watch tidb-cluster up and running:
+
+```
+watch kubectl get pods --namespace default -l app.kubernetes.io/instance=tidb-cluster -o wide
+```
+
+### Test TiDB cluster
+
+There can be a small delay between the pod is up and running, and the service
+is available. You can watch list services available with:
+
+```
+watch kubectl get svc
+```
+
+When you see `tidb-cluster-tidb` appear, it's ready to connect to TiDB server.
+
+After first, forward local port to tidb port:
+
+```
+kubectl port-forward svc/tidb-cluster-tidb 4000:4000
+```
+
+In another terminal, connect to TiDB server with MySQL client:
+
+```
+mysql -h 127.0.0.1 -P 4000 -uroot
+```
+
+Or run SQL command directly:
+
+```
+mysql -h 127.0.0.1 -P 4000 -uroot -e 'select tidb_version();'
+```
+
+### Delete TiDB cluster
+
+```
+helm delete --purge tidb-cluster
+kubectl delete pvc -l app.kubernetes.io/managed-by=tidb-operator # clean old PVs
```

Comment:
The PV reclaim policy is set to `Retain` by tidb-operator by default. So delete PVC would not delete the related PV.

Reply:
`kubectl get pv -l app.kubernetes.io/instance=tidb-cluster -o name | xargs -I {} kubectl patch {} -p '{"spec":{"persistentVolumeReclaimPolicy":"Delete"}} # update reclaim policy of PVs used by tidb-cluster to Delete` is added here to update reclaim policy of PVs.